.golangci.yml

```diff
@@ -0,0 +1,30 @@
+run:
+  deadline: 5m
+
+issues:
+  # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
+  max-issues-per-linter: 0
+  # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
+  max-same-issues: 0
+
+
+linters-settings:
+  govet:
+    check-shadowing: true
+  goling:
+    min-confidence: 0
+  gocyclo:
+    min-complexity: 25
+  maligned:
+    suggest-new: true
+  dupl:
+    threshold: 100
+  goconst:
+    min-len: 2
+    min-occurrences: 2
+
```

Comment:
I'd also like something along the lines of
```yaml
  lll:
    line-length: 160
```

Reply:
okay but I want 100 and the default is 120. which is the maximum I am willing to go to, for two reasons:
1. I don't think 100 characters is all that little in the fast majority of cases and if you are going over that chances are the code is already unreadable you don't need an extra 60 characters (I suppose you don't have 20 indentations which is a problem in of itself, and lll counts tab as 1 char). Yes in some cases - mostly table test cases this is .. the more readable way although even there I can argue that figuring a way for it to be more concise will add to the readability and probably will be easier to extend/write in the future. but 160 is obscene IMO
2. On my laptop in my vim, which takes all the horizontal space, a single line is a grand total of 207 characters (and you know how big my font is). But I always have a vertical split primarily because I need to read some other piece of code or am constantly switching between them. This is why my lines go red after the 100 character in my vim. 
Is one or two lines being wrapped every now and again the worst thing - nope but it also doesn't help much - yes I can read it but sometimes it is not easy to move along those lines or figuring out if this is a new line or is the third wrapped line in a row. 
 When this is rare it is okay - I can live with the idea that looking over a bunch of table test cases half of them will span multiple lines on my screen, but if we raise this for the whole code it will mean there will be what a lot of people will write. 120 is too much in both cases but without it I am of the opinion a lot of our current code base crosses the boundary and am pretty sure adding `//nolint` to all of them when we touch them will become tedious. But for 120 per line it looks okay: after [this commit](https://github.com/loadimpact/k6/pull/943/commits/dc880890b34111966c915591fe873c5987f988d2) is 142 with the tests 53 without them. If I  set it to 160 we get to 24 with tests and 5 without them. 

IMO leaving it as it is and re-evaluating it after we get a few warnings from this and it turns out that rewriting the code will be worse and adding `//nolint` takes way too much time. 
p.s. The majority of the long lines in non tests are flag definitions and struct tags as far as I can see. And of course there are a few string concats that should probably be fmt.Sprintfs

Reply:
OK, :handshake: for leaving it for the default of 120. Not sure why, but I thought it was 80... I can live with 120 and adding `//nolint` where it makes sense. 

I definitely don't like to read long lines of dense code either, but for test cases and other boring stuff, long lines are frequently far more readable than splitting them apart into multiple lines. But for actual code, even 100 is likely way too much...